CONTRIBUTING.md

```diff
@@ -0,0 +1,154 @@
+
+## Committing Rules
+
+First, the commits. They should always look like the following description:
+
+Action Verb (Add, Delete, Fix ...) + Title + Description of what you have done + Related Issue Number
+
+This should look like this:
+
+```bash
+    git commit -m "Add new rules to the CONTRIBUTING.md
+
+    Add Committing Rules #2"
+```
+
+You should always commit frequently and based on the given exemple.
+
+If possible, please certify your commits (-S).
+
+## Testing Rules
+
+When something can be verified by a test, we will be looking for tests in your PR.
+
+To compile your tests, the corresponding CMakeLists.txt should look like this:
+
+```CMake
+    set (SRC_ROOT ${PROJECT_SOURCE_DIR}/tests/unit_tests)
+
+    set (TEST_SRC_ROOT
+        ${SRC_ROOT}/very_basic_test.cpp
+    )
+
+    ## Documented Example ##
+    create_test(
+        ## First, the name of the test ##
+        TEST_NAME very_basic_test
+        ## Second, the sources for the test ##
+        TEST_SRC ${TEST_SRC_ROOT}
+        ## Last, the depenancies (necessary libraries) ##
+        ## In this case none, otherwise add TEST_DEPS ${DEPENDANCY_NAME}
+    )
+```
+
+And the file with all the unit tests will look like this:
+
+``` C++
+    #include <criterion/criterion.h>
+
+    /// @brief Very Basic Test
+    Test(Basic, some_basics_tests)
+    {
+        cr_assert_eq(42, 42);
+    }
+```
+
+For more details, you can go from the project's root to tests/unit_tests directory.
+
+If you're in functionnal tests, please add a bash/bat script to execute them.
+
+The lack of test in your code can be one of the reasons of refusing a PR.
+
+## Pull Request Rules
+
+### The creator
+
+When your work is done on a branch, you must make a pull request.
+
+The name is the same for the branch.
+However, you must write a description as a reminder of what you have done.
+It may be all about the issue, but often you have unexpected fixes.
+It must follow the following template :
+
+```md
+# Purpose:
+The purpose of this pull request is to add the service John.
+
+## Added features:
+- This service has been added to the application. It's include:
+    - A action which is used to ...
+    - A reaction which does this ...
+
+## Updated features:
+- This part has been changed because the new feature needed to be implemented this way.
+
+## Fixed features:
+- This features has been fixed.
+- This one too.
+
+## Removed features:
+- This features has been removed because it was unnecessary.
+```
+
+Of course, if a section is empty, it can be removed from the description.
+
+### The reviewers
+
+For the reviewers, prioritize the people who can really help you with your work.
+
+A reviewer must read, understand and test
+
+You must have at least 2 reviewers who approve your pull request.
+
+Try not to exceed 24 hours before reading your PRs.
+
+It is obviously forbidden to approve a merge until all the reviewers have accepted the PR as well as the success of the workflows.
```

Comment:
Maybe add a conclusion part with ours contact ?

Reply:
Will be done in the README.md